### DISCUSSION TITLE
Accessing Form State

### DISCUSSION THREAD

**Discussion by MattHeslington** [svecosystem/formsnap#188] [Q&A] [answered]
Hello,

How would one access the (reactive) form state, before it's submitted. Let's say I have a form which allows users to submit data for something they want to sell. I would like to be able to show a live preview of how their listing will look as they fill out the form.
Any help very much appreciated.
Cheers,
Matt

**Comment by huntabyte** [comment, marked as answer]
You can use the `$form.data` from Superforms.

**Reply by MattHeslington** [reply]
Lovely, cheers
